secretsclient/client.go

```diff
@@ -1115,3 +1116,33 @@ func (c *client) FindSSHAuth() (map[string][]string, error) {
 	}
 	return foundKeys, nil
 }
+
+func (c *client) RedeemOAuthToken(oauthToken string) (string, error) {
+	connectRequest := api.OAuthConnectRequest{
+		Token: oauthToken,
+	}
+	status, body, err := c.doCall("PUT", "/api/v0/oauth/connect", withJSONBody(&connectRequest))
+	if err != nil {
+		return "", err
+	}
+	if status != http.StatusOK {
+		msg, err := getMessageFromJSON(bytes.NewReader([]byte(body)))
+		if err != nil {
+			return "", errors.Wrap(err, fmt.Sprintf("failed to decode response body (status code: %d)", status))
+		}
+		return "", errors.Errorf("failed to connect OAuth login: %s", msg)
+	}
+
+	var connectResponse api.OAuthConnectResponse
+	err = c.jm.Unmarshal(bytes.NewReader([]byte(body)), &connectResponse)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to unmarshal OAuth connect response")
+	}
+
+	email, err := c.SetLoginToken(connectResponse.Token)
```

Comment:
It seems like the auth header would eventually get written as. `Authorization: <jwt>`, rather than `Authroization: Bearer <jwt>`. 

We could just prepend the string `Bearer ` here for the time being. 

I think the whole auth system here could use a bit of touching up to better adopt JWT. I'm looking at that as part of (ET-29)[https://linear.app/earthly-technologies/issue/ET-29/use-jwt]